### PR TITLE
Update link_credential_phishing_voicemail_language.yml

### PR DESCRIPTION
**Pull request by zoomequipd**
# Description

1) Adjust initial regex check to remove "or" and instead inspect all 3 fields within the any
2) Adjust max link limit 
3) remove check for display_name from "2 of" conditions, which was already checked once as part of the initial regex
4) add "listen" to keywords

# Associated samples

FN that now matches
- [Sample 1](https://platform.sublime.security/messages/21a5b2b1aadabcbbdbdc7b0049787c940677edff66347b4ba57b025a6e17108a)

### CODE REVIEW

**Comment by zoomequipd** [comment]
pulled in #1823

**Comment by zoomequipd** [comment]
/mql-mimic-exempt:, 269282, 319824, 409756

269282 , 319824, 409756 = actually flags in analyzer

File: detection-rules/link_credential_phishing_voicemail_language.yml [hunk 1]
```diff
@@ -60,15 +58,15 @@ source: |
       any(body.links,
           regex.contains(.display_text, '[^a-z]*[A-Z][^a-z]*')
           and regex.icontains(.display_text,
-                              '(voice|audio|call|missed|caii)(\s?|-)(mail|message|recording|call|caii)|transcription|open mp3|audio note'
```
Comment (aidenmitchell) [inline detection-rules/link_credential_phishing_voicemail_language.yml:63]: Do we mean to drop `audio note` here?

File: detection-rules/link_credential_phishing_voicemail_language.yml [hunk 2]
```diff
@@ -60,15 +58,15 @@ source: |
...
+                              '(v[nm]|voice|audio|call|missed|caii)(\s?|-)(mail|message|recording|call|caii)|transcription|open mp3|playback|\([0-9]{3}\).(\*\*\*|[0-9]{3}).\*\*\*'
           )
       )
     ),
     (
       any(body.links,
           .href_url.path == "/ctt"
           and regex.icontains(.display_text,
-                              '(voice|audio|call|missed|caii)(\s?|-)(mail|message|recording|call|caii)|transcription|open mp3|audio note'
```
Comment (aidenmitchell) [inline detection-rules/link_credential_phishing_voicemail_language.yml:71]: Same here, do we mean to remove `audio note`?

**Review by morriscode — APPROVED** [review]

**Pull request merged**